[Entity Analytics] [Entity Store] Run init requests sequentially to prevent resource exists error

tiansivive opened the pull request:
## Summary

This PR fixes an issue where running init for both `user` and `host` entity engines in parallel would cause a race condition while enabling the risk engine, resulting in a `Resource already exists` error.

tiansivive committed:
savepoint

tiansivive committed:
savepoint

tiansivive committed:
clening up

elasticmachine commented:
Pinging @elastic/security-entity-analytics (Team:Entity Analytics)

tiansivive committed:
Merge branch 'main' into ea-entity-store-bug-10861

tiansivive commented:
FYI: I updated the log messages here because it helped me debug and so I think it's valuable to keep them. In any case, there are upcoming PRs that will improve the log messages even more

lgestc committed:
[Security Solution] Remove index pattern field from Sourcerer (#190113)

## Summary

This PR removes index pattern field from the sourcerer model, replacing
it with direct access to data view spec.

The end goal for the sourcerer is to just utilize platform wide Data
View Types, instead of some custom abstractions / containers such as
indexPattern field which is effectively a DataViewSpec, just packed up
differently.

maxcold committed:
[Cloud Security] add posture type for CSPM dashboard accounts link (#197633)

## Summary

- fixes https://github.com/elastic/security-team/issues/10914

mykolaharmash committed:
[Observability Onboarding] Show search bar even when category is not selected (#197825)

Closes https://github.com/elastic/observability-dev/issues/4065 🔒

Fixes an issue when search bar is not visible unless a category is
selected.

![CleanShot 2024-10-25 at 14 53
44@2x](https://github.com/user-attachments/assets/774d2fe2-e4f0-4a46-a851-a0f756a96b12)

georgianaonoleata1904 committed:
[ResponseOps][Cases] Miscount of total numbers of alerts in telemetry (#196112)

Closes https://github.com/elastic/kibana/issues/177208

## Summary

Problem: 
- the metrics collected in telemetry for alerts don't count the total
number of alerts on a case correctly.

Solution: 
- added new aggregation function: getUniqueAlertCommentsCountQuery,
which is now responsible for defining the cardinality aggregation for
counting unique alert comments by alertId.
- in the aggs section of the savedObjectsClient.find, the new
cardinality aggregation query was added
- the total number of alerts is updated to be the result extracted from
the new aggregation

Example: 

![Screenshot 2024-10-22 at 15 20
40](https://github.com/user-attachments/assets/c418c82e-2e35-4c7f-969d-7f4f25bdbc9d)


- in the telemetry object, we have the following info: 
<img width="331" alt="Screenshot 2024-10-22 at 15 21 40"
src="https://github.com/user-attachments/assets/6419e72d-84b4-4068-a741-6e32c6e966f7">

---------

Co-authored-by: Antonio <antoniodcoelho@gmail.com>

georgianaonoleata1904 committed:
[ResponseOps][Cases]Add instructions of how to create a connector in the create case form (#197041)

Closes https://github.com/elastic/kibana/issues/189246

## Summary

- A helper text was added in the create case form to tell the user that
needs to create a connector in the stack management > cases > settings
before attaching it to a case
- A new "add connector" button was placed in the stack management >
cases > settings page, in the connectors section



https://github.com/user-attachments/assets/7866b41a-11b5-4ca3-bd65-988412ab1e2f

---------

Co-authored-by: Antonio <antoniodcoelho@gmail.com>

jughosta committed:
[Discover][ES|QL] Rename Documents tab to Results (#197833)

## Summary

This PR renames Documents label to Results for ES|QL mode.

<img width="1091" alt="Screenshot 2024-10-25 at 15 44 32"
src="https://github.com/user-attachments/assets/8678bb86-7e4b-4341-9bb3-50becced655b">
<img width="1676" alt="Screenshot 2024-10-25 at 15 52 55"
src="https://github.com/user-attachments/assets/c79d2ee0-62e1-4506-bcb5-29552287f140">




### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

szwarckonrad committed:
[EDR Workflows] Enable Blocklist CY in MKI (#197952)

Since adding this test file was tied to changes in Kibana, we initially
couldn’t enable it in the MKI. The MKI relies on a Kibana image built
from the main branch, and at that time, the necessary changes for these
tests to pass hadn’t yet been merged. Now that these updates are
included in the main branch, the Kibana image used in MKI has the
required changes, so we can proceed with enabling the tests.

Manual MKI run -
https://buildkite.com/elastic/kibana-serverless-security-solution-quality-gate-defend-workflows/builds/1545

vitaliidm committed:
[Security Solution][Detection Engine] removes legacy alerting endpoints from Security Solution dev scripts (#197424)

## Summary

 - addresses https://github.com/elastic/kibana/issues/95842

---------

Co-authored-by: Ryland Herrick <ryalnd@gmail.com>

elena-shostak committed:
[CodeQL] Local run script (#194272)

## Summary

This PR introduces a script that allows developers to run CodeQL
analysis locally. It uses a Docker container with prebuilt CodeQL
queries to facilitate easy setup and execution.
The script has the following key steps:
- Creating a CodeQL database from the source code. The database is
essentially a representation of the codebase that CodeQL uses to analyze
for potential issues.
- Running the analysis on the created database,
`javascript-security-and-quality` suit is used.

### Usage
```
bash scripts/codeql/quick_check.sh -s path/to/your-source-dir
```
For example
```
bash scripts/codeql/quick_check.sh -s ./x-pack/plugins/security_solution/public/common/components/ml/conditional_links
```

The `-s` option allows you to specify the path to the source code
directory that you wish to analyze.

### Why custom Docker file?
Checked the ability to use MSFT image for local run
https://github.com/microsoft/codeql-container. Turned out it has several
problems:
1. The published one has an error with [execute
permissions](https://github.com/microsoft/codeql-container/issues/53).
2. Container has outdated nodejs version, so it didn't parse our syntax
(like `??`) and failed.
3. The technique used in the repository to download the CodeQL binaries
and precompile the queries is outdated in the sense that GitHub now
offers pre-compiled queries you can just download. Follow this
[comment](https://github.com/microsoft/codeql-container/issues/53#issuecomment-1875879512).

Taking this into consideration I have created a lightweight docker image
without extraneous dependencies for go/.net/java.

## Context and interdependencies issues
There are issues sometimes when analyze run returns no results,
particularly when analyzing a single folder.
It might be due to the missing context for the data flow graph CodeQL
generates or context for interdependencies. This is actually a trade off
of running it locally for a subset of source directories. We need to
explicitly state that in the documentation and advise to expand the
scope of source code directories involved for local scan.

Documentation for triaging issues will be updated separately.

__Closes: https://github.com/elastic/kibana/issues/195740__

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/246264)
* Commit: e0053c96955134ea5bd809c6b67a4028a69a3c87

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dc2d8e46345d9afaaa436cbe297f6e4fb9f90981) | [after](https://github.com/elastic/kibana/commit/e0053c96955134ea5bd809c6b67a4028a69a3c87) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 20.2MB | 20.2MB | +69.0B |



<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246264","number":246264,"commit":"e0053c96955134ea5bd809c6b67a4028a69a3c87"}],"number":246264}
buildkite-pr-comment-->

kdelemme committed:
chore(slo): remove tests migrated to agnostic framework (#197711)

Resolves https://github.com/elastic/kibana/issues/183397

## Summary

This PR is a follow up of https://github.com/elastic/kibana/pull/195927,
that removes the old and migrated tests to the agnostic framework.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Dzmitry Lemechko <dzmitry.lemechko@elastic.co>

ymao1 committed:
[Response Ops][Task Manager] change task claiming interface to stop using observables  (#196196)

Resolves https://github.com/elastic/kibana/issues/184952

## Summary

Changing task claimers to return promises instead of observables. This
is a code refactor and should not have any effect on task claiming
functionality.

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

criamico committed:
[Fleet] Fix agents count in agent list table and add tooltip with correct info (#197834)

Fixes https://github.com/elastic/kibana/issues/195441

## Summary
Selection agent count on agent list table gets incorrect when there are
multiple hosted agents, especially if they are on inactive state. In
fact to calculate the selected number of agents we were getting hosted
agents, but without taking into account the filtering applied on the
page, i.e. we were always getting all the hosted agent (inactive too).
This caused the final calculation to be off. 

In this PR I'm fixing [the
query](https://github.com/elastic/kibana/pull/197834/files#diff-9707a4b93a96749876e4cf173a0b39cd5a620e311e2652c5ed4b8670ca7e6becR309-R320)
used to get those agents to take in account the filters and I'm also
adding a small tooltip that breaks up the number of agents (selected,
total, hosted)

### Testing
- Make sure to have many agents, hosted and not in different states
(inactive, unenrolled)
- To make a hosted agent inactive follow the steps explained
[here](https://github.com/elastic/kibana/issues/195441)
- Verify that the selection numbers are correct: select agents on all
pages and hover on the new tooltip shown besides the "selected agents".
This number should match the number shown on the actions dropdown


### Screenshots
<img width="2376" alt="Screenshot 2024-10-25 at 17 00 44"
src="https://github.com/user-attachments/assets/81d2836a-f997-4ccb-a23c-3d2cfbfa62d3">
<img width="2409" alt="Screenshot 2024-10-25 at 17 00 59"
src="https://github.com/user-attachments/assets/ba21933a-f1e6-457e-8059-e87b3e29a7d1">



https://github.com/user-attachments/assets/c153c491-29a1-481c-a3e3-25bab6412963




### Checklist

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

maxcold committed:
[Cloud Security] exclude unknown findings from compliance score calculation (#197829)

## Summary

Findings from 3rd party date can have `result.evaluation: unknown`. This
leads to incorrect posture/compliance score in our flows. This PR
removes these findings from the score calculation and graphical
representation. properly introducing `unknown` in the compliance score
UX flows will be solved separately

- fixes https://github.com/elastic/security-team/issues/10913

### Screenshots

<img width="1473" alt="Screenshot 2024-10-25 at 14 19 03"
src="https://github.com/user-attachments/assets/c69e45b0-7da1-4eb8-b83a-f895e7b7c3a4">



### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

SiddharthMantri committed:
[Authz] OAS Descriptions for Route Authz (#197001)

Closes https://github.com/elastic/kibana/issues/191714

## Summary

Update process router to generate authz descriptions based on the new
Route Security objects.


### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

mikecote committed:
Add tags to connector run failures indicating if it's user or framework error (#197818)

Resolves https://github.com/elastic/kibana/issues/197315

In this PR, I'm adding the following tags to the connector failure logs
so it makes it easier to filter for systematic errors.

- `connector-run-failed` for logs specific to connector run failures
- `user-error` for errors caused by the user
- `framework-error` for systematic errors

## To verify

You can either use the jest test to observe the returned flags or set
your logging to JSON and make connectors fail.

kibana.yml to set logging to JSON
```
logging:
  appenders:
    json-layout:
      type: console
      layout:
        type: json
  root:
    appenders: [json-layout]
```

tomsonpl committed:
[EDR Workflows] Fix Cypress tests failing on Alerts step (#197384)

jaredburgettelastic committed:
Delete data when clearing security entity store (#197938)

Fixed a bug where the "Clear all entities" button in the security entity
store didn't delete data due to a missing query parameter.

pheyos committed:
FTR  SAML Auth - Adjust stateful internal request header (#197994)

## Summary

This PR adds the `x-elastic-internal-origin` header to the stateful
internal request headers used by FTR.
This fixes an issue that we're seeing when running deployment agnostic
tests against ESS on 9.0.0-SNAPSHOT.

wayneseymour committed:
[SKIP ON MKI] reporting datastream (#197958)

## Summary

See details: https://github.com/elastic/kibana/issues/197955

wayneseymour committed:
[SKIP ON MKI] discover reporting (#197959)

# Summary

See details: https://github.com/elastic/kibana/issues/197957

alexwizp committed:
fix: [Stateful:Connectors:New connector page]Configuration form missing instructions and field names from announcement (#197963)

Closes: #197586

## Description
Forms, requiring user input, should have clear instructions on how to
fill them. Specific fields can have their own help (guidance) text on
how to fill them with examples.
All fields which are present in the form can be programmatically
determined, especially for the users using assistive technology to
understand what fields are present, what input is expected.

## What was changed: 
1. `aria-label` values ​​are explicitly set for
`ConnectorConfigurationField` child components. I suspect that due to
the dynamic nature of this component, the standard mechanism does not
work properly.


## Screen: 

<img width="1163" alt="image"
src="https://github.com/user-attachments/assets/00e1bd87-30b3-4c8f-a3d7-0c7774028a66">

drewdaemon committed:
[ES|QL] detect the type of `COUNT(*)` (#197914)

## Summary

We weren't properly detecting the type of the expression `COUNT(*)`. Now
we are!

Before:
<img width="950" alt="Screenshot 2024-10-25 at 4 38 08 PM"
src="https://github.com/user-attachments/assets/e9bd8d78-d0c8-4069-a818-5bf3486b925b">

After:
<img width="1093" alt="Screenshot 2024-10-25 at 4 35 44 PM"
src="https://github.com/user-attachments/assets/235c63dc-7d6c-49df-9adf-e225c4550a42">


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

drewdaemon committed:
[ES|QL] remove signatures from function suggestion labels (#197842)

## Summary

Showing a single acceptable function signature in the suggestions list
is confusing since it may imply that other options are not accepted.
Instead, this PR follows the Typescript tooling and simply shows the
function name in the list.

![Screenshot 2024-10-25 at 8 17
38 AM](https://github.com/user-attachments/assets/9caf4998-b144-45d4-8a53-b41846714d5b)

The signatures are still available in the details flyout.


<img width="981" alt="Screenshot 2024-10-25 at 8 50 01 AM"
src="https://github.com/user-attachments/assets/17ca7b55-9c88-4a42-91e0-762cfc4809b5">




### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

ymao1 committed:
[Response Ops][Task Manager] Propagate `msearch` error status code so backpressure mechanism responds correctly (#197501)

Resolves https://github.com/elastic/response-ops-team/issues/240

## Summary

Creating an `MsearchError` class that preserves the status code from any
msearch errors. These errors are already piped to the managed
configuration observable that watches for and responds to ES errors from
the update by query claim strategy so I updated that filter to filter
for msearch 429 and 503 errors as well.

## To Verify

1. Make sure you're using the mget claim strategy
(`xpack.task_manager.claim_strategy: 'mget'`) and start ES and Kibana.
2. Inject a 429 error into an msearch response.

```
--- a/x-pack/plugins/task_manager/server/task_store.ts
+++ b/x-pack/plugins/task_manager/server/task_store.ts
@@ -571,6 +571,8 @@ export class TaskStore {
     });
     const { responses } = result;

+    responses[0].status = 429;
+
     const versionMap = this.createVersionMap([]);
```

3. See task manager log the msearch errors and eventually reduce polling
capacity

```
[2024-10-23T15:35:59.255-04:00][ERROR][plugins.taskManager] Failed to poll for work: Unexpected status code from taskStore::msearch: 429
[2024-10-23T15:35:59.756-04:00][ERROR][plugins.taskManager] Failed to poll for work: Unexpected status code from taskStore::msearch: 429
[2024-10-23T15:36:00.257-04:00][ERROR][plugins.taskManager] Failed to poll for work: Unexpected status code from taskStore::msearch: 429
[2024-10-23T15:36:00.757-04:00][ERROR][plugins.taskManager] Failed to poll for work: Unexpected status code from taskStore::msearch: 429
...

[2024-10-23T15:36:06.267-04:00][WARN ][plugins.taskManager] Poll interval configuration is temporarily increased after Elasticsearch returned 19 "too many request" and/or "execute [inline] script" error(s).
[2024-10-23T15:36:06.268-04:00][WARN ][plugins.taskManager] Capacity configuration is temporarily reduced after Elasticsearch returned 19 "too many request" and/or "execute [inline] script" error(s).
```

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

Commit:
[Stateful sidenav] Fix dashboard listing breadcrumbs (#197986)

thomheymann committed:
[Onboarding] Check for user privileges before creating an API key (#197964)

Resolves https://github.com/elastic/kibana/issues/192983

Check for user privileges before creating an API key

<img width="1413" alt="Screenshot 2024-10-28 at 10 28 58"
src="https://github.com/user-attachments/assets/aa54ce74-98ac-43f0-b422-ab3d895c97ab">

nchaulet committed:
[Fleet] Couple agent and package policies spaces (#197487)

ersin-erdal committed:
Consolidate Rule schemas (#195613)

Towards: #172513

This PR removes `RawRule` type from `alerting/server/type` and
`RuleAttributes` schema/types.
And uses the `RawRule` that is used for ModelVersions instead of them.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

ryankeairns committed:
Improve header button spacing (#197896)

Closes #197873 

## Summary

- Fix spacing between 'Give feedback' and AI button.
- While in the neighborhood, I noticed the search button had some extra
padding, so I fixed that quick too.
- Lastly, I pushed the project switcher to the far right. This is only
used for local development of Serverless and was otherwise appearing
between production buttons.

### Before
<img width="420"
src="https://github.com/user-attachments/assets/f5abe1af-1762-4658-8040-d802b9752667"
/>

<img width="420"
src="https://github.com/user-attachments/assets/04288ff3-6012-4518-866f-0dea4ad62401"
/>

### After
<img width="420"
src="https://github.com/user-attachments/assets/ef2e0eff-d4f4-4ceb-bbf8-c39d2e3e9949"
/>

## QA
To test, you'll need to run this in serverless mode where the 'Give
feedback' button appears.
You can force it on by setting the following line equal to `true`.

https://github.com/elastic/kibana/blob/ae9c0d385015f3068a04af46678e18e2f00b519a/src/plugins/guided_onboarding/public/plugin.tsx#L50

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

vitaliidm committed:
[Security Solution][Detection Engine] fixes preview logs issue when switching between rules (#197098)

## Summary

 - addresses https://github.com/elastic/kibana/issues/196616

markov00 committed:
[Lens] fix showing points on line by default (#197828)

## Summary

Fix the missing style to show points in line charts. Line charts should
show data points whenever the distance between points is larger than
40px.

criamico committed:
[Fleet] Fix flaky agentless test (#197951)

Fixes https://github.com/elastic/kibana/issues/189038

## Summary

Attempt to fix [this flaky test
](https://buildkite.com/elastic/kibana-on-merge/builds/53472#0192c57d-51ca-4b9b-a934-dc13b0b9b7ca)
failing with
```

Timed out in waitForNextUpdate after 1000ms.
--
  |  
  | at waitForNextUpdate (node_modules/@testing-library/react-hooks/lib/core/asyncUtils.js:96:13)

```
I'm adding a longer timeout hoping to resolve this issue. There is no
way to run jest tests with flaky test runner so I'm not sure how to
verify that the flakiness is really gone.

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

pheyos committed:
Serverless tests - enable dot-prefixed index validation (#197141)

## Summary

This PR enables Elasticsearch dot-prefixed index validation for
serverless tests.

jennypavlova committed:
[ObsUx][Infra] Remove no longer used feature flags for GA features (#197684)

Closes #197612 
## Summary

This PR removes the feature flags
(`observability:enableInfrastructureHostsView` and
`enableInfrastructureContainerAssetView` ) for host and container views

Feature settings under `Infrastructure > Settings` (ignore the profiling
one)
| Before | After |
| ------ | ----- |

|![image](https://github.com/user-attachments/assets/f7ee9585-e96c-4492-9116-8de3c5e0c9c6)
|
![image](https://github.com/user-attachments/assets/1c75e229-bd29-45c5-a354-cfdcb324ea9d)
|

## Testing:
The feature flags should not appear in the settings
The Host menu item should be visible
The Container asset views should be always enabled and visible (without
an option to disable them)
The Host asset views should be always enabled and visible (without an
option to disable them)




https://github.com/user-attachments/assets/a4d574d0-2669-4f96-88a4-a40c2f5023c2

nreese committed:
do not set full screen mode on ExitFullScreenButton re-render (#198012)

### Background
https://github.com/elastic/kibana/pull/194892 is refactoring
[DashboardRenderer](https://github.com/elastic/kibana/blob/3391344e8dc8377d359b918521b6c48838cde8ae/src/plugins/dashboard/public/dashboard_container/external_api/dashboard_renderer.tsx)
component to replace Dashboard Embeddable with a plain old javascript
object. Dashboard Embeddable rendered its contents in a new react tree.
The new implementation does not. Since the new implementation does not
render the dashboard in a new react tree, any re-render in
`DashboardViewport` parent components causes `ExitFullScreenButton` to
re-render. In its current form, re-rendering `ExitFullScreenButton`
calls `onExit`, which causing dashboard to exit full screen mode.

This PR makes use of `useCallback` to fix the issue where re-rending
`ExitFullScreenButton` calls `onExit`.

### Test steps
1) Open dashboard that ships with sample web logs data set
2) switch to view mode
3) click "Full screen" button
4) Maximize a panel. Verify dashboard stays in full screen mode.

jbudz committed:
[ci] Run linting before tests (#197310)

Linting is a frequent source of build failures. By increasing the cpu
count we can run this check before starting our highly-parallel tests
without impacting total build time.

legrego committed:
Fix documentation for session lifespan default (#198065)

This pull request includes an update to the
`docs/settings/security-settings.asciidoc` file to clarify the default
session lifespan settings for different installation environments.

Documentation update:

*
[`docs/settings/security-settings.asciidoc`](diffhunk://#diff-97a4c4e3696b33b246f55ddd794608530b693f0a7a66ae1361a32b67c7461523L204-R204):
Clarified that the default session lifespan is 30 days for on-prem
installations and 24 hours for Elastic Cloud installations.

wayneseymour committed:
[FTR][Ownership] Assign aiops, custom branding, etc (#197468)

## Summary

Assign test files to small number of reviewers

### Assignment Reasons

Assigned custom_branding due to
https://github.com/elastic/kibana/blob/main/x-pack/plugins/custom_branding/kibana.jsonc#L4

Assigned response_ops_docs due to the name

Assigned monitoring due to
https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/kibana.jsonc#L4

Assigned so managment due to
https://github.com/elastic/kibana/blob/main/src/plugins/saved_objects_management/kibana.jsonc#L4

Assigned aiops due to
https://github.com/elastic/kibana/blob/main/x-pack/plugins/aiops/kibana.jsonc#L4

Assigned banners_functional due to Pierre being all over the git blame.
:lol:

Assigned x-pack/test/screenshot_creation due to
https://github.com/elastic/kibana/pull/197468#discussion_r1817460031

Contributes to: https://github.com/elastic/kibana/issues/194817

e40pud committed:
[Security GenAI][BUG] KB index entry created via pdf upload does not give the right response (#198020)

## Summary

These changes fix the issue with the wrong response of the AI Assistant
using knowledge base tool and index entry generated from a PDF file.

The issue happens because we are using the first chunk of uploaded PDF
document as a context that we pass to LLM instead of using inner hits
chunks which are actual parts of the document relevant to the questions.

Here is [the blog
post](https://www.elastic.co/search-labs/blog/semantic-text-with-amazon-bedrock)
that talks about the strategy of using inner hits to get the most
relevant documents. (see `Strategy 1: API Calls` section)

### Upload + index PDF

1. Navigate to Integrations page
2. Select "Upload a file"
3. Select and upload a PDF file
4. Press Import button
5. Switch to Advanced tab
6. Fill in "Index name"
7. Add additional field > Add semantic text field > Fill in form
  * Field: `attachment.content`
  * Copy to field: `content`
  * Inference endpoint: `elser_model_2`
8. Press Add button
9. Press Import button

### Add KB index entry (with uploaded PDF data)

1. Navigate to AI Assistant's Knowledge Base page
2. New > Index
3. Fill in "New index entry" form (below are main fields)
  * Name: `[add entry name]`
  * Index: `[select index name created during uploading a PDF file]`
  * Field: `content`
4. Press Save button

### Testing notes

Enable knowledge base feature via

```
xpack.securitySolution.enableExperimental:
  - 'assistantKnowledgeBaseByDefault'
```

### Example PDF for testing

**PDF document**:
[Elastic Global Threat Report
2024](https://github.com/user-attachments/files/17544720/elastic-global-threat-report-2024.pdf)

**KB Index entry**:
Data Description: "Use this tool to answer questions about the Elastic
Global Threat Report (GTR) 2024"
Query Instruction: "Key terms to return data relevant to the Elastic
Global Threat Report (GTR) 2024"

**Questions**:
1. Who are the authors of the GTR 2024?
2. What is the forecast for the coming year in GTR 2024?
3. What are top 10 Process Injection by rules in Windows endpoints in
GTR 2024?
4. What is the most widely adopted cloud service provider this year
according to GTR 2024?
6. Give a brief conclusion of the GTR 2024

**Current behaviour**:

<img width="656" alt="Screenshot 2024-10-28 at 16 43 48"
src="https://github.com/user-attachments/assets/90615356-8807-4786-b58d-ca28c83aaec9">

**Fixed behaviour**:

<img width="655" alt="Screenshot 2024-10-28 at 16 44 47"
src="https://github.com/user-attachments/assets/9ebefbcc-20c2-4c79-98f3-11fa6acf3da6">

TinaHeiligers committed:
Improves pattern matching for data telemetry (#197876)

## Summary

Addresses concerns with Regex matching.

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

shahzad31 committed:
[Synthetics] Refactor delete route !! (#195387)

## Summary

Fixes https://github.com/elastic/kibana/issues/193790 !!

Refactor delete route !! 

Make sure to send delete response in bulk to synthetics service !!

nchaulet committed:
[Fleet] Prevent hosted policies space change (#198043)

rshen91 committed:
Revert "[Canvas] Update kbn/flot to remove table.replace() issue" (#198067)

Fixes #197998

Reverts elastic/kibana#195643

kibanamachine committed:
skip failing test suite (#181466)

tsullivan committed:
[Global Search] Instantly set `isLoading=true` when search value changes (#197750)

## Summary

Close https://github.com/elastic/kibana/issues/77059

This PR solves the bug by setting the `isLoading` flag outside of the
block of debounced code whenever the search term changes.

This also makes a few slight cleanups to `search_bar.tsx`, which is
quite large. I avoided doing any serious cleanups that would make the
diff hard to read or detract from the fix.

mistic committed:
skip flaky suite (#178404)

mistic committed:
skip flaky suite (#197335)

mistic committed:
skip flaky suite (#189038)

mistic committed:
skip flaky suite (#192126)

mistic committed:
skip flaky suite (#197765)

mistic committed:
skip flaky suite (#189739)

mistic committed:
skip flaky suite (#196766)

mistic committed:
skip flaky suite (#174661)

jaredburgettelastic committed:
Added redirect option after Entity CSV upload (#197937)

## Summary

Added a redirect button to view Entities after a successful Asset
Criticality CSV file upload process

Additionally, made some small changes to the copy for the Entity Store
management workflows

<img width="1490" alt="Screenshot 2024-10-27 at 9 49 17 PM"
src="https://github.com/user-attachments/assets/77b587b7-8300-40ae-adc2-5119aa5f39ab">

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

elastic-renovate-prod[bot] committed:
Update dependency msw to ^2.4.12 (main) (#198060)

This PR contains the following updates:

| Package | Type | Update | Change | Pending |
|---|---|---|---|---|
| [msw](https://mswjs.io) ([source](https://togithub.com/mswjs/msw)) |
devDependencies | patch | [`^2.4.11` ->
`^2.4.12`](https://renovatebot.com/diffs/npm/msw/2.4.11/2.4.12) |
`2.5.2` (+3) |

---

### Release Notes

<details>
<summary>mswjs/msw (msw)</summary>

### [`v2.4.12`](https://togithub.com/mswjs/msw/releases/tag/v2.4.12)

[Compare
Source](https://togithub.com/mswjs/msw/compare/v2.4.11...v2.4.12)

#### v2.4.12 (2024-10-21)

##### Bug Fixes

- **node:** preserve headers instanceof when recording raw headers
([#&#8203;2321](https://togithub.com/mswjs/msw/issues/2321))
([`a58a300`](https://togithub.com/mswjs/msw/commit/a58a300687a48e13c0268403a71183cf7825f748))
[@&#8203;paoloricciuti](https://togithub.com/paoloricciuti)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://togithub.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy40MjUuMSIsInVwZGF0ZWRJblZlciI6IjM3LjQyNS4xIiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJUZWFtOkNsb3VkIFNlY3VyaXR5IiwiYmFja3BvcnQ6c2tpcCIsInJlbGVhc2Vfbm90ZTpza2lwIl19-->

Co-authored-by: elastic-renovate-prod[bot] <174716857+elastic-renovate-prod[bot]@users.noreply.github.com>

Ikuni17 committed:
[CI] Disable UpdateCLI workflow on forks (#196624)

## Summary

The UpdateCLI workflow is running against forks when it shouldn't be:

https://github.com/Ikuni17/kibana/actions/runs/11359905941

Ikuni17 committed:
Fix Dev Container KBN_DIR (#195810)

## Summary

In #193488, `KBN_DIR` was changed to be a dynamic variable. It wasn't
being properly propagated through the build process in the Dockerfile
and the full path to `env.sh` wasn't being set. This passes the
directory as a build `ARG` as well to fix the path.

kibanamachine committed:
[api-docs] 2024-10-29 Daily api_docs build (#198103)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/875

maximpn committed:
[Security Solution][Hotfix] Avoid blocking prebuilt rule upgrade upon conflicts with disabled feature flag (#198106)

## Summary

It turned out some of the update rule buttons are disabled. This is a side effect of the functionality not fully hidden under a feature flag. This PR hides prebuilt rule customisation functionality disabling update rule buttons under `prebuiltRulesCustomizationEnabled` feature flag.

## Before

![image](https://github.com/user-attachments/assets/b7ca5ff8-be37-47a7-ad7e-b85386909f38)

## After

<img width="1719" alt="image" src="https://github.com/user-attachments/assets/349223dc-dda5-46fb-832f-d7097a81580e">

<img width="1721" alt="image" src="https://github.com/user-attachments/assets/a28512f6-e605-460e-884d-571ab408a7d9">

jennypavlova committed:
[Infra] Fix anomalies flyout navigation failing test (#197999)

Closes #192882 

The issue was that the url was checked too early which resulted in
checking the locator url instead of the page url after navigating so I
added a check for loading before the URL check and this solved the
issue.


https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7284

jennypavlova committed:
[ObsUx][Infra] Unskip and try to fix hosts view test (#197861)

Closes #191806

The parts of the unskipped test were fixed here - the only case I
couldn't find the reason for failing after unskipping it is `should have
an option to open the chart in lens` - for some reason, the whole menu
is gone when checking the CI and I couldn't reproduce that so this will
be the only skipped part so we can at least have all the other `Hosts
view` tests

CohenIdo committed:
[Cloud Security] Fix flaky metering tests

flash1293 committed:
[Observability Onboarding] Set tech preview badges correctly (#197831)

As discussed, adjust the first onboarding page:
* Remove tech preview labels from EA flows
* Change wording

<img width="857" alt="Screenshot 2024-10-25 at 15 23 47"
src="https://github.com/user-attachments/assets/ceffbe90-019d-4fa8-ab6f-16cbf0aaf3d4">
<img width="814" alt="Screenshot 2024-10-25 at 15 23 59"
src="https://github.com/user-attachments/assets/4f5cf657-6fd2-479f-aa92-9460bc2ecfef">

tomsonpl committed:
[EDR Workflows] Skip Osquery test in MKI (#198117)

gergoabraham committed:
[EDR Workflows] Improve on unavailable shard exception flakiness in cypress (#197864)

## Summary

The cypress task `cy.task('indexEndpointHosts')` sometimes throws
`no_shard_available_action_exception`, when transforms are stopped. This
looks like a temporary issue, and in other tests it is simply retried.

This PR adds the retry logic for this type of error, and unskips some
tests.

closes #194135
closes #191914

### Checklist

Delete any items that are not applicable to this PR.
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed

maxcold committed:
[Cloud Security] add vulnerabilties data set to filter for links from CNVM dashboard (#197648)

## Summary

- fixes https://github.com/elastic/security-team/issues/10915

tiansivive committed:
[Entity Analytics] [Entity Store] Telemetry (#196880)

This PR adds telemetry for the Entity Store.
Client side tracks UI enablement actions, whilst Kibana side tracks
execution time of the store initialisation process and execution time of
the enrich policy task.
Finally we also track number of entities in the store

tiansivive committed:
fixing conflicts
